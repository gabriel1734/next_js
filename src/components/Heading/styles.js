import styled, { css } from 'styled-components';

const titleSize = {
  small: (theme) => css`
    font-size: ${theme.fonts.sizes.medium};
  `,
  medium: (theme) => css`
    font-size: ${theme.fonts.sizes.large};
  `,
  big: (theme) => css`
    font-size: ${theme.fonts.sizes.xlarge};
  `,
  huge: (theme) => css`
    font-size: ${theme.fonts.sizes.xhuge};
    ${mediaFont(theme)};
  `,
};

const mediaFont = (theme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.fonts.sizes.xlarge};
    background-color: 'red';
  }
`;

export const Title = styled.h1`
  ${({ theme, colorDark, size, uppercase }) => css`
    color: ${colorDark ? theme.colors.white : theme.colors.primaryColor};
    ${titleSize[size](theme)};
    text-transform: ${uppercase ? 'uppercase' : 'none'};
  `}`;
