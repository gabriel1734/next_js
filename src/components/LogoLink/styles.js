import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    color: ${theme.colors.primaryColor};
    text-decoration: none;
    

    > img {
      height: 3rem;
    }
  `}
`;
