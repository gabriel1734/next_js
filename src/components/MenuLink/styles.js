import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    display: block;
    text-decoration: none;
    font-size: ${theme.fonts.sizes.medium};
    padding: ${theme.spacings.small};
    color: ${theme.colors.primaryColor};
    position: relative;
    

    &::after {
      content: '';
      position: absolute;
      bottom: 0.76rem;
      left: 50%;
      width: 0%;
      height: 0.2rem;
      background-color: ${theme.colors.secondaryColor};
      transition: all 0.5s ease-in-out;
    }
    &:hover::after {
      left: 25%;
      width: 50%;
    }
  `}
`;
