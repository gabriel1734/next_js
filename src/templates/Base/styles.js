import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${() => css`
    padding-top: 2rem;
  `}
`;
