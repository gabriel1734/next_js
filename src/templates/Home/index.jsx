import { useEffect, useRef, useState } from 'react';
import { Base } from '../Base';
import { mapData } from '../../api/map-data';
import { PageNotFound } from '../PageNotFound';
import { Loading } from '../../components/Loading';
import { GridTwoColumn } from '../../components/GridTwoColumn';
import { GridContent } from '../../components/GridContent';
import { GridText } from '../../components/GridText';
import { GridImage } from '../../components/GridImage';
import { useLocation } from 'react-router-dom';
import config from '../../config';

function Home() {
  const [data, setData] = useState([]);
  const location = useLocation();
  const isMounted = useRef(true);

  useEffect(() => {
    const pathName = location.pathname.replace(/[^a-z0-9-_]/gi, '');
    const slug = pathName ? pathName : config.defaultSlug;
    const load = async () => {
      try {
        const data = await fetch(
          `http://localhost:1337/api/pages?slug=${slug}&populate=deep&pagination[pageSize]=1&sort[0]=id:desc`,
        );
        const json = await data.json();
        const pageData = mapData([json.data[0].attributes]);
        setData(pageData[0]);
      } catch (e) {
        setData(undefined);
      }
    };
    if (isMounted.current === true) {
      load();
    }
    return () => {
      isMounted.current = true;
    };
  }, [location]);

  useEffect(() => {
    if (data === undefined) {
      document.title = `Página não encontrada | ${config.SiteName}`;
    }

    if (data && !data.slug) {
      document.title = `Carregando... | ${config.SiteName}`;
    }

    if (data && data.slug) {
      document.title = `${data.title} | ${config.SiteName}`;
    }
  }, [data]);

  if (data === undefined) {
    return <PageNotFound />;
  }

  if (data && !data.slug) {
    return <Loading />;
  }

  const { menu, sections, footerHtml, slug } = data;
  const { links, text, link, srcImg } = menu;

  return (
    <Base
      links={links}
      footerHtml={footerHtml}
      LogoData={{ text, link, srcImg }}
    >
      {sections.map((section, index) => {
        const { component } = section;
        const key = `${slug}-${index}`;
        if (component === 'section.section-two-columns') {
          return <GridTwoColumn key={key} {...section} />;
        }
        if (component === 'section.section-content') {
          return <GridContent key={key} {...section} />;
        }
        if (component === 'section.section-grid') {
          return <GridText key={key} {...section} />;
        }
        if (component === 'section.section-grid-image') {
          return <GridImage key={key} {...section} />;
        }
      })}
    </Base>
  );
}

export default Home;
