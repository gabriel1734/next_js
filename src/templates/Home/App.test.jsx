import Home from '.';
import { renderTheme } from '../../styles/render-theme';

describe('Home', () => {
  it('should render correctly', () => {
    const { container } = renderTheme(<Home />);
    expect(container).toMatchSnapshot();
  });
});
