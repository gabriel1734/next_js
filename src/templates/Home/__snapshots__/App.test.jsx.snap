// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render correctly 1`] = `
.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
  width: 100%;
}

.c3 {
  color: #0A1128;
  font-size: 6.2rem;
  text-transform: uppercase;
}

.c5 {
  font-size: 2.4rem;
}

.c2 {
  text-align: center;
  max-width: 58rem;
  margin: 0 auto;
}

.c4 {
  margin: 6.2rem 0;
}

.c0 {
  background-color: #fff;
  color: #0A1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media (max-width:768px) {
  .c3 {
    font-size: 4rem;
    background-color: 'red';
  }
}

<div>
  <div
    class="c0"
    id=""
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h2
          class="c3"
        >
          [ Error ] 404
        </h2>
        <div
          class="c4"
        >
          <div
            class="c5"
          >
            <p>
              A página não foi encontrada! 
              <a
                href="/"
              >
                Clique aqui para voltar
              </a>
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
